src/common/filesystembase.cpp

```diff
@@ -869,21 +864,20 @@ bool FileSystem::setAclPermission(const QString &unsafePath, FolderPermissions p
         newAceIndex++;
     }
 
+#if 0
+    // TODO: figure out if this is necessary; before 4.0.2 this did not work at all
+    //       --> currentFolder used to be the parent directory of `path`, so this ended up to be
+    //           iterating over nonexisting entries inside `path`.
+    //           due to the early `return` in the loop this may have never reached the call to
+    //           what-used-to-be `SetFileSecurityW` (now `SetSecurityInfo`)
     if (safePathFileInfo.isDir() && applyAlsoToFiles) {
         const auto currentFolder = QDir{path};
         const auto childFiles = currentFolder.entryList(QDir::Filter::Files);
         for (const auto &oneEntry : childFiles) {
             const auto childFile = joinPath(path, oneEntry);
-
             const auto rawChildFile = reinterpret_cast<const wchar_t *>(childFile.utf16());;
-            const auto attributes = GetFileAttributesW(rawChildFile);
-
-            // testing if that could be a pure virtual placeholder file (i.e. CfApi file without data)
-            // we do not want to trigger implicit hydration ourself
-            if ((attributes & FILE_ATTRIBUTE_SPARSE_FILE) != 0) {
-                continue;
-            }
 
+            // apparently as long as we do not attempt to read the file contents we will not trigger an implicit hydration ourselves
```

Comment:
we now prevent self-hydration with a proper setting when setting up the cloud file provider connection